core/controllers/base.py

```diff
@@ -184,6 +184,8 @@ def __init__(self, request, response):  # pylint: disable=super-init-not-called
         self.user = current_user_services.get_current_user()
         self.user_id = current_user_services.get_user_id(
             self.user) if self.user else None
+        self.role = user_services.get_user_role_from_id(
```

Comment:
@seanlip .... current_user_services is an appengine thing, right. Could you explain what is it's main task?